### PR TITLE
feat(scala): parse indented match, indented blockExpr, `using` and splatted args, and `enum`

### PR DESCRIPTION
**Pull request by brandonspark**
## What:
This PR allows us to parse indented matches in Scala, like
```
object bar:
  private def foo(x: Int) =
    x match
      case p if p < 15 => "a"
      case p if p < 30 => "b"
      case p if p < 45 => "c"
      case _           => "d"
```

## Why:
parse rate!

## How:
Added a `inBracesOrIndented`

## Test plan:
Parse rate to `0.9460688054925819` instead of `0.941790402456788`

The enclosed test case parses to
```
Pr(
  [DefStmt(
     ({
       name=EN(
              Id(("foo", ()),
                {id_info_id=1; id_hidden=false; id_resolved=Ref(None); id_type=Ref(None); id_svalue=Ref(None); }));
       attrs=[]; tparams=[]; },
      ClassDef(
        {ckind=(Object, ()); cextends=[]; cimplements=[]; cmixins=[]; cparams=[
         ];
         cbody=[F(
                  DefStmt(
                    ({
                      name=EN(
                             Id(("foo", ()),
                               {id_info_id=2; id_hidden=false; id_resolved=Ref(
                                None); id_type=Ref(None); id_svalue=Ref(
                                None); }));
                      attrs=[KeywordAttr((Private, ()))]; tparams=[]; },
                     FuncDef(
                       {fkind=(Method, ());
                        fparams=[Param(
                                   {pname=Some(("x", ())); pdefault=None;
                                    ptype=Some({t_attrs=[];
                                                t=TyN(
                                                    Id(("Int", ()),
                                                      {id_info_id=4; id_hidden=false; id_resolved=Ref(
                                                       None); id_type=Ref(
                                                       None); id_svalue=Ref(
                                                       None); }));
                                                });
                                    pattrs=[];
                                    pinfo={id_info_id=3; id_hidden=false; id_resolved=Ref(
                                           Some((Parameter, -1))); id_type=Ref(
                                           None); id_svalue=Ref(None); };
                                    })];
                        frettype=None;
                        fbody=FBExpr(
                                StmtExpr(
                                  Block(
                                    [DefStmt(
                                       ({
                                         name=EN(
                                                Id(("x", ()),
                                                  {id_info_id=5; id_hidden=false; id_resolved=Ref(
                                                   None); id_type=Ref(
                                                   None); id_svalue=Ref(
                                                   None); }));
                                         attrs=[KeywordAttr((Const, ()))]; tparams=[
                                         ]; },
                                        VarDef({vinit=Some(L(Int((Some(2), ())))); vtype=None; })));
                                     Return((), Some(L(Int((Some(5), ())))), ())])));
                        }))))];
         })))])
```
which is truly a `blockExpr`, so I am satisfied.

Closes PA-2687

PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date
- [X] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: stats/parsing-stats/lang/scala/projects.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@ https://github.com/akka/akka
 https://github.com/yahoo/CMAK
 https://github.com/ornicar/lila
 https://github.com/gitbucket/gitbucket
-https://github.com/twitter/finagle
+# https://github.com/twitter/finagle
```
Comment (nmote) [inline stats/parsing-stats/lang/scala/projects.txt]: Could you justify this or remove the change?

**Review by nmote — APPROVED** [review]
Seems legit other than the removal of a project from the parsing stats

File: stats/parsing-stats/lang/scala/projects.txt (hunk 1 above)
Comment (brandonspark) [reply stats/parsing-stats/lang/scala/projects.txt]: that test causes a segfault on my machine, i usually comment it for testing purposes and occasionally i forget and accidentally commit it (that's what happened here)

Comment (nmote) [reply stats/parsing-stats/lang/scala/projects.txt]: That's unsettling

**Comment by aryx** [comment]
seems similar to another PR I just accepted.

**Pull request merged**